### PR TITLE
appease mypy in qa app

### PR DESCRIPTION
**Pull request by fvankrieken**

### CODE REVIEW

File: dcpy/utils/s3.py [hunk 1]
```diff
@@ -13,6 +19,16 @@
     TimeRemainingColumn,
 )
 
+ACL = Literal[
```
Comment (fvankrieken) [inline dcpy/utils/s3.py:22]: This was to make the new s3 typing happy

File: dcpy/utils/s3.py [hunk 2]
```diff
@@ -2,9 +2,15 @@
...
-from typing import Any, Optional
+from typing import Any, Literal, TYPE_CHECKING
 import boto3
+from botocore.response import StreamingBody
 from botocore.client import Config
+
+if TYPE_CHECKING:
+    from mypy_boto3_s3.client import S3Client
+else:
+    S3Client = object
```
Comment (fvankrieken) [inline dcpy/utils/s3.py:13]: This was sort of interesting. Makes sense though - for s3 typing we need the additional `boto-stubs` package with the `s3` option. Nice to not have to include an extra package in our build environments, only dev.

**Comment by fvankrieken** [comment]
This ended up being a bit meatier than I expected, mainly because I wanted to fix up a few things in dcpy and that led down a rabbit hole. A couple little things to point out

- started going with `str | None` instead of `Optional`
- using `dict` and `list` instead of `Dict` and `List` for type annotations. We currently have a mix of both. You used to have to use the latter to specify subtype (`List[str]`) but as of 3.9 the lowercase are fine (and built-in - uppercase must be imported from typing)

File: dcpy/utils/s3.py (hunk 2 above)
Comment (damonmcc) [reply dcpy/utils/s3.py:13]: are there any other typeing-related imports we'd also wanna put this condition around?

Comment (fvankrieken) [reply dcpy/utils/s3.py:13]: Not that I can see

**Review by damonmcc — APPROVED** [review]
nice! probably wanna squash on merge?

**Pull request merged**